### PR TITLE
Add doc for LieTensor multiplication __mul__, __matmul__, and @

### PR DESCRIPTION
**Pull request by AAkashyap1**

### CODE REVIEW

File: pypose/lietensor/basics.py [hunk 1]
```diff
@@ -122,6 +122,86 @@ def add(input, other, alpha=1):
...
+
+    where :math:`\bm{x}` is the ``input`` LieTensor, :math:`\bm{a}` is the ``other`` value,
+    and :math:`\bm{y}` is the output value.
+    
+    :obj:`LieTensor` @ :obj:`LieTensor` -> :obj:`LieTensor` (group multiplication) 
+        
+    :obj:`LieTensor` @ :obj:`Tensor` -> :obj:`Tensor` (group action)
+
+    Args:
+        input (:obj:`LieTensor`) the input LieTensor (Lie Group or Lie Algebra).
```
Comment (wang-chen) [inline pypose/lietensor/basics.py:141]: format not correct, you need `:` between type and its explanation.
for example:
```
input (:obj:`LieTensor`): the input LieTensor (Lie Group or Lie Algebra).
```

File: pypose/lietensor/basics.py [hunk 2]
```diff
@@ -122,6 +122,86 @@ def add(input, other, alpha=1):
...
+
+        other (:obj:`Tensor` or :obj:`LieTensor`): the value for input to be multiplied by.
+
+    Return:
+        the product with the same type as ``other``
```
Comment (wang-chen) [inline pypose/lietensor/basics.py:146]: You need a type for returns. For example,
```
Tensor/LieTensor: the product with the same type as ``other``.
```
Please refer to other docs for function return.

File: pypose/lietensor/basics.py [hunk 3]
```diff
@@ -122,6 +122,86 @@ def add(input, other, alpha=1):
...
+    .. math::
+        \bm{y}_i =
+        \bm{x}_i \times \bm{a}_i
+
+    where :math:`\bm{x}` is the ``input`` LieTensor, :math:`\bm{a}` is the ``other`` value,
+    and :math:`\bm{y}` is the output value.
+    
+    :obj:`LieTensor` @ :obj:`LieTensor` -> :obj:`LieTensor` (group multiplication) 
+        
+    :obj:`LieTensor` @ :obj:`Tensor` -> :obj:`Tensor` (group action)
```
Comment (wang-chen) [inline pypose/lietensor/basics.py:138]: You need to clearly define different situations, refer to the doc of [Exp](https://pypose.org/docs/main/generated/pypose.Exp/)

**Review by wang-chen — CHANGES_REQUESTED** [review]
I think you are misuse some symbols:
1. `matmul` only refers to the `@` operation, `mul` refers to the `*` operation.
2. in equations, `x` should only refer to `@`.
You also need to clearly discuss the different cases.

For example,
- [x] LieTensor * scalar
- [x] LieTensor * Tensor (last dimension is 3)
- [x] LieTensor * Tensor (last dimension is 4)
- [ ] LieTensor * LieTensor
- [ ] LieTensor @ LieTensor

File: pypose/lietensor/basics.py (hunk 1 above)
Comment (wang-chen) [reply pypose/lietensor/basics.py:141]: if some of the cases, you may also let the readers refer to doc of `Act`: https://pypose.org/docs/main/generated/pypose.Act/

File: pypose/lietensor/basics.py [hunk 4]
```diff
@@ -123,6 +123,347 @@ def add(input, other, alpha=1):
...
+        .. math::
+            \bm{x} = [t_x, t_y, t_z, q_x, q_y, q_z, q_w, s],
+        .. math::
+            \bm{x} = [t_x', t_y', t_z', q_x', q_y', q_z', q_w', s'],
+        
+        Peforms same calculations as with :obj:`RxSO3_type` to calculate the 
+        quaternion and scaling factor of the product :math:`(\bm{x} \ast \bm{a})_{q, s}`,
+        using an SE3 type instead of an SO3 type. Uses the same calculations as with 
+        :obj:`SE3_type` to calculate the translational vector of the product 
+        :math:`(\bm{x} \ast \bm{a})_{t}`.
```
Comment (wang-chen) [inline pypose/lietensor/basics.py:250]: This is not a doc, you need to give explicit equation.

**Comment by wang-chen** [comment]
I just made several changes, please have a look @AAkashyap1 , you also need to add 3 more examples for `RxSO3`, `SE3`, and `Sim3`.

**Comment by AAkashyap1** [comment]
@wang-chen, I added in the examples and your changes looked good.

**Review by wang-chen — APPROVED** [review]

**Pull request merged**